src/blop/utils/__init__.py

```diff
@@ -17,13 +17,28 @@ def get_beam_stats(image: np.ndarray, threshold: float = 0.5) -> dict[str, float
     fim = sp.ndimage.median_filter(fim, size=3)
     fim = sp.ndimage.gaussian_filter(fim, sigma=1)
 
-    m = fim > threshold * fim.max()
+    m = fim > (threshold * fim.max())
     area = m.sum()
+    if area == 0.0:
```

Comment:
why is this needed?

Reply:
When I had the simulated beamline without noise and the beam was off-screen, the total sum of the image that is above the threshold of 0.5 was 0.0. This leads to a divide by zero error a few lines down. Since the image is all the same value, there is no beam to extract and we can return 0.0 for everything.